core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateSetGenericAggregation.java

```diff
@@ -41,14 +70,55 @@ public static void input(@AggregationState HyperLogLogState state, @SqlType(Stan
     }
 
     @InputFunction
-    public static void input(@AggregationState HyperLogLogState state, @SqlType(StandardTypes.BIGINT) long value)
+    @TypeParameter("T")
+    public static void input(
+            @OperatorDependency(
+                    operator = XX_HASH_64,
+                    argumentTypes = "T",
+                    convention = @Convention(arguments = NEVER_NULL, result = FAIL_ON_NULL))
+                    MethodHandle methodHandle, // ignored but engine requires all matching input methods signatures if @TypeParameter is used
+            @AggregationState HyperLogLogState state,
+            @SqlType("T") long value)
     {
         HyperLogLog hll = getOrCreateHyperLogLog(state);
         state.addMemoryUsage(-hll.estimatedInMemorySize());
         hll.add(value);
         state.addMemoryUsage(hll.estimatedInMemorySize());
     }
 
+    @InputFunction
+    public static void input(@AggregationState HyperLogLogState state, @SqlType("BOOLEAN") boolean value)
+    {
+        HyperLogLog hll = getOrCreateHyperLogLog(state);
+        state.addMemoryUsage(-hll.estimatedInMemorySize());
+        hll.add(value ? 1 : 2);
```

Comment:
is it practical?

i think we should do something like 7569071af22708e4e600c4197720e489d86ac781

Reply:
Not sure I can do better if I have to return hll to the user at the end of the day.

Reply:
you're right.

will such two-value HLL be exact? can we choose values so that it's exact?

Reply:
Yeah - those guarantee exact count. And it is covered by TestApproximateSetBoolean